### PR TITLE
carts: Fix item collection

### PR DESCRIPTION
**Pull request by sfan5**
reported here: https://forum.minetest.net/viewtopic.php?p=374446#p374446

### CODE REVIEW

**Comment by paramat** [comment]
Just from curiosity, why is the requirement for physical_state now the opposite?

**Comment by sfan5** [comment]
The code that checks for `physical_state` was added with the [initial merge of carts](https://github.com/minetest/minetest_game/commit/1e691c442907d6bb5602bc34fdbb3c9a7436c223#diff-9812a7ca8951c9d004454c5c7c339f40R321) in 0.4.15.
The builtin items however didn't actually have a `physical_state` at that point, until [this engine commit](https://github.com/minetest/minetest/commit/c9ec1a8580038549040b36c1f201f60c59d92d83) (5.1.0) added it.

So in conclusion it's been broken since (and including) 5.1.0.
